src/js/providers/default.ts

```diff
@@ -0,0 +1,186 @@
+import { PLAYER_STATE, MEDIA_TYPE } from 'events/events';
+import { GenericObject } from 'types/generic.type';
+
+const noop: () => void = function(): void { /* noop */ };
+const returnFalse: () => boolean = (() => false);
+const getNameResult: { name: string } = { name: 'default' };
+const returnName: () => { name: string } = (() => getNameResult);
+
+type SeekRangeType = {
+    start: number;
+    end: number;
+};
+
+type SourceType = {
```

Comment:
seems like there's probably more that goes into this, should i move to `generic.type.ts`?